.github/workflows/build_artifacts.yaml

```diff
@@ -0,0 +1,188 @@
+        run: rustup target add $TARGET
+      - name: Setup AArch64 Tool
+        if: ${{ matrix.target == 'aarch64-unknown-linux-gnu' }}
+        working-directory: bindings/c
+        run: |
+          sudo apt update
+          sudo apt install gcc-aarch64-linux-gnu -y
+          echo "CC=aarch64-linux-gnu-gcc" >> $GITHUB_ENV
+          mkdir .cargo
+          cat << EOF > .cargo/config.toml
+          [target.aarch64-unknown-linux-gnu]
+          linker = "aarch64-linux-gnu-gcc"
+          EOF
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+      - name: Setup Service
+        working-directory: bindings/c
+        env:
+          OPENDAL_FEATURES: "layers-blocking,services-${{ matrix.service }}"
+        run: |
+          python -m pip install toml
+          cat << EOF > ./setup_feature.py
+          import toml
+          import os
+
+          with open('Cargo.toml', 'r') as f:
+              cargo_toml = toml.load(f)
+
+          enabled_features = os.environ.get('OPENDAL_FEATURES', '').split(',')
+
+          opendal_features = []
+          for feature in enabled_features:
+              if feature.strip():
+                  opendal_features.append(feature.strip())
+
+          cargo_toml['dependencies']['opendal']['features'] = opendal_features
+
+          with open('Cargo.toml', 'w') as f:
+              toml.dump(cargo_toml, f)
+          EOF
+          python ./setup_feature.py
+      - name: Build ${{ matrix.service }} ${{ matrix.target }}
+        working-directory: bindings/c
+        env:
+          SERVICE: ${{ matrix.service }}
+          TARGET: ${{ matrix.target }}
+        run: |
+          CC="${{ env.CC }}" cargo build --target $TARGET  --release
+          sudo apt install zstd
+          zstd -22 ./target/$TARGET/release/libopendal_c.so -o ./libopendal_c.so.$TARGET.zst
+      - uses: actions/upload-artifact@v4
+        with:
+          name: "libopendal_c_${{ inputs.opendal_tag }}_${{ matrix.service }}_${{ matrix.target }}"
+          if-no-files-found: "error"
+          path: "bindings/c/libopendal_c.so.${{ matrix.target }}.zst"
+          overwrite: "true"
+  generate:
+    needs: [ matrix, build ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+      - uses: actions/download-artifact@v4
+      - name: Generate Template
+        env:
+          MATRIX: ${{ needs.matrix.outputs.matrix }}
+          VERSION: ${{ inputs.opendal_tag }}
+        working-directory: generate
+        run: |
+          go run generate.go
+      - name: Auto Commit
+        env:
+          VERSION: ${{ inputs.opendal_tag }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global user.name "Github Actions"
+          git config --global user.email "actions@github.com"
+          git branch opendal-$VERSION
+          git checkout opendal-$VERSION
+          git add -A
```

Comment:
Can we just push to `main` and create tag over it? I feel like it's easier to checkout and use.

Reply:
Sure.